frontend/src/metabase/query_builder/components/filters/modals/InlineCategoryPicker/InlineCategoryPicker.tsx

```diff
@@ -75,6 +75,10 @@ export function InlineCategoryPickerComponent({
       });
   }, [dimension, safeFetchFieldValues, shouldFetchFieldValues]);
 
+  const showInlinePicker =
+    fieldValues.length <= MAX_INLINE_CATEGORIES &&
+    (!filter || filter?.operatorName() === "=");
```

Comment:
nit: I was a bit confused at first with this condition. I think rearranging the condition a bit would tell the story better. e.g.
```suggestion
    (fieldValues.length <= MAX_INLINE_CATEGORIES &&
      filter?.operatorName() === "=") ||
    !filter;
```

It reads
- Show inline picker if
  - `fieldValues` is less than `MAX_INLINE_CATEGORIES` **AND** filter operator is `=`
    **OR**
  -  there is no `filter` (empty state)

Rather than
- Show inline picker if
  - `fieldValues` is less than `MAX_INLINE_CATEGORIES`
  **AND**
  -  there is no `filter` (empty state) **OR** filter operator is `=`
 
When there's no `filter` it should imply no `fieldValues` (length = 0).

What do you think?

Reply:
That changes the logic though. If there's no filter, we still need to check whether there are too many categories to show. No matter what the operator or filter state is, we don't show inline checkboxes if there are more than `MAX_INLINE_CATEGORIES`.

Reply:
You're right. I was confused.